From 249414dbbe87d67dd399deab26229acb159a6ed8 Mon Sep 17 00:00:00 2001
From: Wei-Chiu Chuang <weichiu@cloudera.com>
Date: Mon, 2 May 2016 06:43:15 -0700
Subject: [PATCH 1505/2748] HDFS-10186. DirectoryScanner: Improve logs by
 adding full path of both actual and expected
 block directories.  Contributed by Rakesh R

(cherry picked from commit 262d769b4d31d03aa4bf78ef9ebb5774154d5a64)

Change-Id: I1791fa88d658c3d78d35d31779d542379abfe7cb
---
 .../hdfs/server/datanode/DirectoryScanner.java     |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
index 3fffd95..0bda256 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DirectoryScanner.java
@@ -891,8 +891,7 @@ public ScanInfoPerBlockPool call() throws IOException {
             break;
           }
         }
-        verifyFileLocation(blockFile.getParentFile(), bpFinalizedDir,
-            blockId);
+        verifyFileLocation(blockFile, bpFinalizedDir, blockId);
         report.add(new ScanInfo(blockId, blockFile, metaFile, vol));
       }
       return report;
@@ -902,12 +901,15 @@ public ScanInfoPerBlockPool call() throws IOException {
      * Verify whether the actual directory location of block file has the
      * expected directory path computed using its block ID.
      */
-    private void verifyFileLocation(File actualBlockDir,
+    private void verifyFileLocation(File actualBlockFile,
         File bpFinalizedDir, long blockId) {
       File blockDir = DatanodeUtil.idToBlockDir(bpFinalizedDir, blockId);
-      if (actualBlockDir.compareTo(blockDir) != 0) {
+      if (actualBlockFile.getParentFile().compareTo(blockDir) != 0) {
+        File expBlockFile = new File(blockDir, actualBlockFile.getName());
         LOG.warn("Block: " + blockId
-            + " has to be upgraded to block ID-based layout");
+            + " has to be upgraded to block ID-based layout. "
+            + "Actual block file path: " + actualBlockFile
+            + ", expected block file path: " + expBlockFile);
       }
     }
 
-- 
1.7.9.5

